contracts/BC_fusion/StakeHub.sol

```diff
@@ -1015,7 +1016,11 @@ contract StakeHub is System, Initializable {
     function _jailValidator(Validator storage valInfo, uint256 jailUntil) internal {
         // keep the last eligible validator
         bool isLast = (numOfJailed >= _validatorSet.length() - 1);
-        if (isLast) {
+        // 0x08 is the staking channel id. If this channel is closed, then BC-fusion is finished and we should keep the last eligible validator here
+        if (
+            !ICrossChain(CROSS_CHAIN_CONTRACT_ADDR).registeredContractChannelMap(VALIDATOR_CONTRACT_ADDR, 0x08)
```

Comment:
We should put `isLast` before the channel check, because in most cases `isLast` will be false, then channel check will be unnecessary, it can save gas.